plugins/com.google.cloud.tools.eclipse.sdk.ui/src/com/google/cloud/tools/eclipse/sdk/ui/preferences/CloudSdkPreferenceArea.java

```diff
@@ -75,32 +91,96 @@ public void widgetSelected(SelectionEvent event) {
+      updateNow.setText(Messages.getString("UpdateNow")); //$NON-NLS-1$
+
+      customSdkRadio = new Button(parent, SWT.RADIO);
+      customSdkRadio.setText(Messages.getString("CustomSdk")); //$NON-NLS-1$
+    }
+
+    customSdkSubArea = new Composite(parent, SWT.NONE);
+    sdkLocation = new CloudSdkDirectoryFieldEditor(PreferenceConstants.CLOUD_SDK_PATH,
+        Messages.getString("SdkLocation"), customSdkSubArea); //$NON-NLS-1$
     Path defaultLocation = getDefaultSdkLocation();
     if (defaultLocation != null) {
       sdkLocation.setFilterPath(defaultLocation.toFile());
     }
     sdkLocation.setPreferenceStore(getPreferenceStore());
     sdkLocation.setPropertyChangeListener(wrappedPropertyChangeListener);
-    GridLayoutFactory.fillDefaults().numColumns(sdkLocation.getNumberOfControls())
-        .generateLayout(fieldContents);
 
+    if (CloudSdkManager.managedSdkFeatureEnabled()) {
+      GridLayoutFactory.fillDefaults().numColumns(2)
+          .extendedMargins(IDialogConstants.LEFT_MARGIN, 0, 0, 0)
+          .generateLayout(managedSdkSubArea);
+      GridLayoutFactory.fillDefaults().numColumns(sdkLocation.getNumberOfControls())
+          .extendedMargins(IDialogConstants.LEFT_MARGIN, 0, 0, 0)
+          .generateLayout(customSdkSubArea);
+    } else {
+      GridLayoutFactory.fillDefaults().numColumns(sdkLocation.getNumberOfControls())
+          .generateLayout(customSdkSubArea);
+    }
     GridLayoutFactory.fillDefaults().generateLayout(contents);
 
     Dialog.applyDialogFont(contents);
     return contents;
   }
 
+  @VisibleForTesting
+  void loadSdkManagement(boolean loadDefault) {
+    IPreferenceStore preferenceStore = getPreferenceStore();
+    String value;
+    if (loadDefault) {
+      value = preferenceStore.getDefaultString(PreferenceConstants.CLOUD_SDK_MANAGEMENT);
+    } else {
+      value = preferenceStore.getString(PreferenceConstants.CLOUD_SDK_MANAGEMENT);
+    }
+
+    boolean managed = CloudSdkManagement.MANAGED.name().equals(value);
+    managedSdkRadio.setSelection(managed);
+    customSdkRadio.setSelection(!managed);
+  }
+
+  private void updateControlEnablement() {
+    boolean managed = managedSdkRadio.getSelection();
+    autoUpdateCheck.setEnabled(managed, managedSdkSubArea);
+    updateNow.setEnabled(managed);
+    sdkLocation.setEnabled(!managed, customSdkSubArea);
+  }
+
   @Override
   public void load() {
+    if (CloudSdkManager.managedSdkFeatureEnabled()) {
+      loadSdkManagement(false /* loadDefault */);
+      autoUpdateCheck.load();
+      updateControlEnablement();
+    }
     sdkLocation.load();
-    fireValueChanged(VALUE, "", "");
+    fireValueChanged(VALUE, "", ""); //$NON-NLS-1$ //$NON-NLS-2$
   }
 
   @Override
   public void loadDefault() {
+    if (CloudSdkManager.managedSdkFeatureEnabled()) {
+      loadSdkManagement(true /* loadDefault */);
+      autoUpdateCheck.loadDefault();
+      updateControlEnablement();
+    }
```

Comment:
I wonder if loadDefault() should also have a `fireValueChanged()`.

Reply:
Currently `fireValueChanged()` is only about `sdkLocation`, and since I'm going to remove `sdkLocation.loadDefault()`, we won't fire the value changed event.